mlir/test/Dialect/Vector/vector-transfer-flatten.mlir

```diff
@@ -356,3 +356,15 @@ func.func @fold_unit_dims_entirely(%arg0 : vector<8xi32>,
 // CHECK:           %[[VAL_3:.*]] = arith.muli %[[VAL_0]], %[[VAL_1]] : vector<8xi32>
 // CHECK:           %[[VAL_4:.*]] = arith.addi %[[VAL_3]], %[[VAL_2]] : vector<8xi32>
 // CHECK:           return %[[VAL_4]] : vector<8xi32>
+
+// -----
+
+// This test is to make sure there is no crash for empty stride.
```

Comment:
Why not add a proper test instead? That would defend against any future breakage and also test one more quite relevant case.

Reply:
I am sorry but I don't understand. What do you mean by a proper test? This test is a simplification of the original test that was breaking. Without my change it will break the same way, and and with my change it will not crash and run to completion. The comment was added to show why the test was added. I also added some CHECK tags with the tests, to make sure it is executing.

Reply:
The “check” lines added, unlike for other tests in this file, do not reveal/check/verify/document the expected behaviour for this case. Well, apart from making sure that the transformation doesn’t crash. 

Tl,Dr  Please add more check lines.

Reply:
Thanks for explaining it for me! I added more checks. I also added a check-not for for the error message string that the crash would cause.